src/routes/exercise-finished/ExerciseFinishedScreen.tsx

```diff
@@ -142,43 +143,50 @@ const ExerciseFinishedScreen = ({ navigation, route }: Props): ReactElement => {
   const { message, resultColor, buttonText, ResultIcon, navigationAction } = helper()
 
   return (
-    <SafeAreaView style={{ flex: 1 }}>
-      <Root>
-        <UpperSection unlockedNextExercise={unlockedNextExercise}>
-          <Icon onPress={() => navigation.dispatch(closeExerciseAction)}>
-            {unlockedNextExercise ? (
-              <CloseIcon width={wp('6%')} height={wp('6%')} />
-            ) : (
-              <CloseIconWhite width={wp('6%')} height={wp('6%')} />
-            )}
-          </Icon>
-          <ResultIcon width={wp('10%')} height={wp('10%')} />
-          <MessageContainer>
-            <Message unlockedNextExercise={unlockedNextExercise}>{message}</Message>
-            <Results color={resultColor}>
-              {correctResults.length} {labels.results.of} {results.length} {labels.general.words}{' '}
-              {labels.results.correct}
-            </Results>
-            <Progress.Bar
-              color={resultColor}
-              progress={percentageOfCorrectResults}
-              unfilledColor={theme.colors.background}
-              width={wp('40%')}
-              height={wp('2%')}
-              borderWidth={0}
-            />
-          </MessageContainer>
-        </UpperSection>
+    <>
+      <RouteWrapper
```

Comment:
Please remove second `SafeAreaView` and use the `RouteWrapper` as a wrapper around the content. Tested it, works fine.